https://redhat.atlassian.net/browse/ACM-31308 separating more files part 3 (#8884)
open-cluster-management/rhacm-docs: observability/obs_add_uw_metrics.adoc

:_mod-docs-content-type: PROCEDURE

[#add-uw-metrics]
= Adding user workload metrics

[role="_abstract"]

> To collect custom workload metrics from managed clusters, create  `ConfigMap` resource named `observability-metrics-custom-allowlist` in a namespace.

Author: ```suggestion
To collect custom workload metrics from managed clusters, create  `ConfigMap` resource named `observability-metrics-custom-allowlist` in a namespace.

After creating the `ConfigMap` resource, the observability collector collects and pushes the metrics from the target namespace to the hub cluster.
```

> To collect custom workload metrics from managed clusters, create  `ConfigMap` resource named `observability-metrics-custom-allowlist` in the `open-cluster-management-observability` namespace.

Author: ```suggestion
To collect custom workload metrics from managed clusters, create  `ConfigMap` resource named `observability-metrics-custom-allowlist` in a namespace.
```

> Add user workload metrics to be collected from your managed clusters by configuring a config map with the name `observability-metrics-custom-allowlist` in the namespace where you want to capture the user workload metrics.

Author: ```suggestion
To collect custom workload metrics from managed clusters, create  `ConfigMap` resource named `observability-metrics-custom-allowlist` in the `open-cluster-management-observability` namespace.
```

.Procedure

. Set the configuration on the managed cluster in namespace where you want to capture the user workload metrics. The name must be `observability-metrics-custom-allowlist`.

. The following example monitors the User workload metric `sample_metrics` from the namespace `monitored_namespace`. If you create the configuration in the `open-cluster-management-addon-observability` namespace instead, the metrics are collected from all the namespaces of the managed cluster. See the following example:

[source,yaml]
----
kind: ConfigMap
apiVersion: v1
metadata:
  name: observability-metrics-custom-allowlist
  namespace: <monitored_namespace>
data:
  uwl_metrics_list.yaml:
    names:
      - <sample_metrics>
----
* Add the namespace where you want to capture the metric.

> * Add the value of the config map data in YAML format. Add list of metric names that you want to collect from the `test` namespace to the `names` section. After you create the config map, the observability collector collects and pushes the metrics from the target namespace to the hub cluster.

Author: ```suggestion
* For the `data` parameter, add the value of the config map data in YAML format. List the metric names that you want to collect from the `test` namespace to the `names` section. 
```
